### ISSUE TITLE
uglify fails when run on browserified bundle

### ISSUE THREAD

**Issue by takempf** [bengourley/hot-rld#6]
My build fails on minification when it comes across the the `hot-rld` client script. I guess `uglify` doesn't know what to do with a template string.

First, a question: Why is `client.js` a template string? Why not just let people include it with a `<script>` or `require()`?

Second, here are a couple solutions:

- Make `client.js` a normal JS file (without the template string)
- Add a `prepublish` hook to convert that JS file into an exported string (as it is now, just without the template string)
- Add a `postpublish` hook to get rid of the converted file

OR

Alternatively `client.js` could just be a normal js file which gets included like so:

```js
if (process.env.NODE_ENV === 'development') {
   require('hot-rld/client');
}
```

**Comment by bengourley** [comment]
Hey. So I wouldn't recommend building `hot-rld` in with your bundled JS and I want to discourage loading it with `require`. It doesn't need to be ugilfy'd because it's only for development, so I'd discourage that too.

`client.js` is meant to be a module that generates a string of JS that you can load as the text content of a `<script>` tag.

That said, I'd be happy to modify this file to help it make more sense. The template literals were:

1. so I could do multi-line strings.
2. so I could pass `port` in as an option to change the client js (just realised this isn't hooked up 😱.

My use case rendering on the server (with node) is `<script>${require('hot-rld/client')()}</script>`.

Basically if I change this to a plain JS file it would have to be something like `fs.readFileSync(require.resolve('hot-rld/client'))` and there would be no simple way to configure the port (don't fancy a regex replace, hah!).

Does this make sense to you? I wonder if you could spec out what you are doing on the server so I can properly understand a different POV than my own use case. Let me know!

Cheers 🌟

**Comment by bengourley** [comment]
hah, I confused myself, I was playing around with both these apis:

1. `require('hot-rld/client')`
2. `require('hot-rld/client')(port)`

It's actually 1. at the moment.

I remember now thinking that the port could be configured with a global variable, if necessary. Here's my suggestion:

```
<script>
  window.__HOT_RLD_PORT__ = 1234
  ${require('hot-rld/client')}
</script>
```
